pandas/core/arrays/interval.py

```diff
@@ -500,8 +500,11 @@ def __getitem__(self, value):
 
         # scalar
         if not isinstance(left, ABCIndexClass):
-            if isna(left):
+            if is_scalar(left) and isna(left):
```

Comment:
My intention with this block of code was to only handle scalars here.  Since `self.left` and `self.right` are always indexes, and we're using `__getitem__` on them to get `left`/`right`, my assumption at the time was that `left`/`right` would either always be a scalar or an `Index` (1d), so `not isinstance(left, ABCIndexClass)` would imply scalar (I guess could use `is_scalar` instead).

With this PR it looks like `Index.__getitem__`  can return a scalar, `Index`, or `ndarray` with `ndim `> 1?  With the last case being temporary until we remove this behavior?  Or am I omitting a case where something else could be returned?  If these are the only three cases, could we handle the `ndim` > 1 case separately before this `if` block?

Something like:
```python
left = self.left[value]
right = self.right[value]

# TODO: remove this block when Index.__getitem__ returning ndim > 1 is deprecated
if np.ndim(left) > 1:
    # GH#30588 multi-dimensional indexer disallowed
    raise ValueError(...)

# scalar
if not isinstance(left, ABCIndexClass):
    ....
```

Makes the logic less nested and easier to remove when we go through with the deprecation.

Reply:
Updated so that this raises ValueError for >1dim indexing on IntervalArray